tests/test_flags/test_dry_run.py

```
@@ -0,0 +1,524 @@
+                                # API should NOT be called when no staged changes
+                                mock_api.assert_not_called()
+
+                                # Should exit with 1 when no changes
+                                mock_exit.assert_called_with(1)
+
+    def test_dry_run_with_auto_stage(self):
+        """Test dry run with -a flag for auto-staging."""
+        test_argv = ["git-commitai", "--dry-run", "-a"]
+
+        with patch("sys.argv", test_argv):
+            with patch("subprocess.run") as mock_subprocess:
```

Ruff (0.12.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)